core/lib/spree/permission_sets/default_customer.rb

```diff
@@ -11,6 +11,9 @@ def activate!
         can [:read, :update], Order do |order, token|
           order.user == user || (order.guest_token.present? && token == order.guest_token)
         end
+        cannot :update, Order do |order|
+          order.completed?
+        end
```

Comment:
Why not have two seperate rules? One read and one update rule. You mentioned that in your commit message. What was the reason you did it this way?

Reply:
I started with that in mind and this has been my [first attempt commit](https://github.com/nebulab/solidus/commit/5bba2fbe572433c04ddf615fed4a32a39cf4ef16). Than I saw #547 and investigate a bit more in the cancancan doc and I found [this page in their wiki](https://github.com/CanCanCommunity/cancancan/wiki/Ability-Precedence) which says that rules are logically or'ed and I think it's a good way to avoid duplicate logic or move it into a method which I don't know where to place.

What do you think? I'd go more with this and change the commit message to better reflect its content (thanks for noticing that)

Reply:
For me, it seems the right way to do.

Reply:
It is absolutely fine 👌